packages/compiler/src/core/checker.ts

```diff
@@ -5822,3 +5907,9 @@ const ReflectionNameToKind = {
 } as const;
 
 const _assertReflectionNameToKind: Record<string, Type["kind"]> = ReflectionNameToKind;
+
+enum Related {
+  false = 0,
+  true = 1,
+  maybe = 2,
```

Comment:
I feel like "unknown" describes this situation better than "maybe", maybe.

Reply:
Typescript also has `unknown` so I kept the same naming

from their docs
```
 * Generally, Ternary.Maybe is used as the result of a relation that depends on itself, and
 * Ternary.Unknown is used as the result of a variance check that depends on itself. We make
 * a distinction because we don't want to cache circular variance check results.
```

Reply:
😂 I stand corrected